lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -509,63 +517,72 @@ def compute_puppet_msi_name(host, opts)
           end
         end
 
-        # Installs Puppet Agent and dependencies from msi
+        # Installs Puppet Agent and dependencies from msi on provided host(s).
         #
-        # @param [Host] host The host to install packages on
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
         # @param [Hash{Symbol=>String}] opts An options hash
         # @option opts [String] :version The version of Puppet Agent to install
         # @option opts [String] :win_download_url The url to download puppet from
         #
         # @note on windows, the +:ruby_arch+ host parameter can determine in addition
         # to other settings whether the 32 or 64bit install is used
-        def install_puppet_agent_from_msi_on(host, opts)
-          arch = host.is_x86_64? ? 'x64' : 'x86'
+        def install_puppet_agent_from_msi_on(hosts, opts)
+          block_on hosts do |host|
 
-          # If we don't specify a version install the latest MSI for puppet-agent
-          if opts['version']
-            host['dist'] = "puppet-agent-#{opts[:version]}-#{arch}"
-          else
-            host['dist'] = "puppet-agent-#{arch}-latest"
-          end
+            is_config_32 = true == (host['ruby_arch'] == 'x86') || host['install_32'] || opts['install_32']
```

Comment:
Consistency with the rest of these checks which we sprinkle through the code... and should really be gathered into a method.